File: src/ruby/test_readme_reader.rb
require 'minitest/autorun'
require_relative 'readme_reader'

class TestReadmeReader < Minitest::Test
  def test_read_readme
    temp_file = Tempfile.new("README.md")
    temp_file.write("Sample README content")
    temp_file.rewind

    assert_equal "Sample README content", read_readme(temp_file.path)

    temp_file.close
    temp_file.unlink
  end

  def test_readme_not_found
    assert_raises(RuntimeError) { read_readme("non_existent.md") }
  end
end

File: src/python/test_readme_reader.py
import unittest
import tempfile
from readme_reader import read_readme

class TestReadmeReader(unittest.TestCase):
    def test_read_readme(self):
        with tempfile.NamedTemporaryFile(mode="w+", delete=False) as temp_file:
            temp_file.write("Sample README content")
            temp_file.seek(0)
            self.assertEqual(read_readme(temp_file.name), "Sample README content")

    def test_readme_not_found(self):
        with self.assertRaises(FileNotFoundError):
            read_readme("non_existent.md")

if __name__ == "__main__":
    unittest.main()
